.github/actions/compile-commit/action.yml

```diff
@@ -7,9 +7,23 @@ inputs:
 runs:
   using: composite
   steps:
+    - name: Get the hash of local working tree
+      id: repo-hash
+      shell: bash
+      run: |
+        TREE_HASH=$( git rev-parse HEAD: )
+        echo "tree-hash=${TREE_HASH}" >> $GITHUB_OUTPUT
+    - name: Cache compile-commit action success
+      id: check-compile-commit-success
+      uses: actions/cache@v3
+      with:
+        path: compile-commit-success.txt
+        key: compile-commit-success-${{ steps.repo-hash.outputs.tree-hash }}
```

Comment:
Why does this use tree-hash instead of commit-hash? Probably deserves a comment/detail in commit message. (commit hash helps to manually find the cache entries to invalidate if needed). Maybe we should also add the branch name in the cache key to make it easier to invalidate if needed.